clang/test/3C/mergebodies2.c

```diff
@@ -0,0 +1,40 @@
+// RUN: rm -rf %t*
+// RUN: 3c -base-dir=%S -addcr -output-dir=%t.checked %s %S/mergebodies1.c --
+// RUN: %clang -working-directory=%t.checked -c mergebodies2.c mergebodies1.c
+// RUN: FileCheck -match-full-lines --input-file %t.checked/mergebodies2.c %s
+// RUN: 3c -base-dir=%t.checked -output-dir=%t.convert_again %t.checked/mergebodies2.c %t.checked/mergebodies1.c --
+// RUN: test ! -f %t.convert_again/mergebodies2.c
+
+// This test and its counterpart check for merging functions with bodies,
+// differing number of parameters, renamed parameters, and multiple mains
+// Note that the name added to a parameter without it may differ based on
+// the order of parsing files.
+
+int safeBody(int a, int *b) {
+// CHECK: int safeBody(int a, _Ptr<int> b) _Checked {
+  return a;
+}
+
+int *wildBody() {
+// CHECK: int *wildBody(int x, _Ptr<int> y, _Ptr<_Ptr<int>> z) : itype(_Ptr<int>) {
+  return (int *)1;
+};
+
+int *wildBody(int x, int *y, int **z);
```

Comment:
Here you have the first parameter type `int x` match in this version and in the version in the other file, it's the arity that doesn't match. What happens if the types don't match? I think `mergeDeclaration` should reject that, right? Do we have a test for that?

Reply:
do you mean the base types, like matching `int` with `char`? As far as I can tell, that was never checked, and so it will merge. I didn't have to change anything to merge `int main()` with `void main()`. It may be something we could do. But I think it fits in the category of clang errors that we are free to ignore. Pointer depth is the only thing we check in 3C code.